scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java

```diff
@@ -0,0 +1,131 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package functionaltests.rm;
+
+import functionaltests.utils.*;
+import org.apache.log4j.Logger;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.objectweb.proactive.core.config.ProActiveConfiguration;
+import org.ow2.proactive.resourcemanager.RMFactory;
+import org.ow2.proactive.scheduler.common.job.JobId;
+import org.ow2.proactive.scheduler.common.job.JobInfo;
+import org.ow2.proactive.scheduler.common.job.JobResult;
+import org.ow2.proactive.scheduler.common.job.JobStatus;
+import performancetests.helper.LogProcessor;
+
+import java.io.File;
+import java.net.URL;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+
+
+
+public class TestRMReconnectioWhileRunning extends MultipleRMTBase {
+
+    static final Logger logger = Logger.getLogger(TestRMReconnectioWhileRunning.class);
+
+    private SchedulerTHelper schedulerHelper;
+
+    private static URL runningJob = TestRMReconnectioWhileRunning.class.getResource("/functionaltests/descriptors/Job_20s.xml");
+    private static URL runningJob1 = TestRMReconnectioWhileRunning.class.getResource("/functionaltests/descriptors/Job_20s-1.xml");
+    private static URL runningJob2 = TestRMReconnectioWhileRunning.class.getResource("/functionaltests/descriptors/Job_20s-2.xml");
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        if (TestScheduler.isStarted()) {
+            SchedulerTHelper.log("Killing previous scheduler.");
+            TestScheduler.kill();
+        }
+        initConfigs();
+    }
+
+
+    @Test
+    public void testTaskIsNotStuckRunning() throws Exception {
+        ProActiveConfiguration.load();
+        RMFactory.setOsJavaProperty();
+        schedulerHelper = new SchedulerTHelper(false, true);
+        schedulerHelper.createNodeSource("local", 3);
+        schedulerHelper.getSchedulerInterface().changePolicy("functionaltests.rm.PolicyWhichThrowsExceptions");
+        JobId jobId = schedulerHelper.submitJob(new File(runningJob.toURI()).getAbsolutePath());
+        JobId jobId2 = schedulerHelper.submitJob(new File(runningJob1.toURI()).getAbsolutePath());
+        JobId jobId3 = schedulerHelper.submitJob(new File(runningJob2.toURI()).getAbsolutePath());
+
+        schedulerHelper.waitForEventTaskRunning(jobId, "running_task_for20s");
+
+        Thread.sleep(200000); // sleep enough that all jobs should be able to finish
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)